src/base/QXmppElement.cpp

```diff
@@ -123,10 +123,13 @@ QXmppElement::~QXmppElement()
 
 QXmppElement &QXmppElement::operator=(const QXmppElement &other)
 {
-    other.d->counter.ref();
-    if (!d->counter.deref())
-        delete d;
-    d = other.d;
+    if (this != &other) // self-assignment check
```

Comment:
This check doesn't actually affect correctness, right?

If so, I'm not sure it should be there. The performance improvements might be marginal, if any. Also, note that, say, typical std::string implementations don't do any self-assignment checks in their operator= (at least, I'm not aware of any implementations doing so).